Source/WebKit/UIProcess/WebPageProxyInternals.h

```diff
@@ -339,9 +341,17 @@ struct WebPageProxy::Internals final : WebPopupMenuProxy::Client
     RefPtr<WebDataListSuggestionsDropdown> dataListSuggestionsDropdown;
 
 #if ENABLE(DRAG_SUPPORT)
+    struct DragOperationData {
+        WebCore::DragData dragData;
+        String dragStorageName;
+        SandboxExtension::Handle sandboxExtensionHandle;
+        Vector<SandboxExtension::Handle> sandboxExtensionsForUpload;
+        std::optional<WebCore::FrameIdentifier> lastFrameID;
+    };
     WebCore::IntRect currentDragCaretEditableElementRect;
     WebCore::IntRect currentDragCaretRect;
     WebCore::DragHandlingMethod currentDragHandlingMethod { WebCore::DragHandlingMethod::None };
+    std::optional<DragOperationData> currentDragOperationData;
```

Comment:
It seems like storing this on the WebPageProxy might not be the right scope.  Can this instead be captured in lambdas or sent over IPC?